src/app/core/user/profile/user-profile.service.js

```diff
@@ -23,35 +20,25 @@ const
  * ==========================================================
  */
 
-module.exports.updatePreferences = (id, pref) => {
-	return User.findOne({ _id: id }).then((user) => {
+module.exports.updatePreferences = (_id, pref) => {
+	return User.findOne({ _id }).exec().then((user) => {
 		const preferences = user.preferences || {};
 		Object.assign(preferences, pref);
 
-		return user.update({$set: { preferences: preferences } }).exec();
+		return user.update({$set: { preferences: preferences } });
```

Comment:
exec

I think this is the last one.

@brianghig @jrassa  Should we be looking to make sure *any* time we make an interaction via mongoose, that we include the .exec()? There's a few more in-lineers (specifically ones that look to be in-lined with a Promise.all) that were not .exec()'d in the existing code.

Reply:
Done - Thanks again @tzfx !